### PR TITLE
global: MANIFEST.in

### PR DESCRIPTION
**Pull request by crepererum**
* Adds MANIFEST.in so that data files are installed as well.

Signed-off-by: Marco Neumann <marco@crepererum.net>

### CODE REVIEW

**Comment by jirikuncar** [comment]
can we add "post-create" `check-manifest -c` instead?

**Comment by crepererum** [comment]
In theory yes, but I don't think it would be very smart. The problem is that after the creation, there exists only a very limited set of files. This is because we wanted to create a bootstrap template, not an example creator. So the created MANIFEST.in would miss many file types. Now there are many people who add template files and other data files to their modules. Someone could argue that they then must add the types to the manifest. But after seeing how many problems people have when creating new modules (e.g. understanding how we discover certain parts, how to create a view/api/...), I think we should make it as easy as possible. This way developers can focus on there real job.

**Comment by jirikuncar** [comment]
@crepererum I'm keeping this on hold as MANIFEST.in should include only existing files, hence encouraging people to run `check-manifest -c` is IMHO preferable. We can add `check-manifest` to Travis so we are sure that nobody forgets to update the file.

--
see also https://gist.github.com/jirikuncar/654560a5e4f5498ffa4f#file-gistfile1-txt-L25

File: {{ cookiecutter.project_shortname }}/.travis.yml [hunk 1]
```diff
@@ -63,6 +63,7 @@ before_script:
   - "inveniomanage database create --quiet || echo ':('"
 
 script:
+  - "check-manifest -c"
```
Comment (jirikuncar) [inline {{ cookiecutter.project_shortname }}/.travis.yml]: you want to run just the "check" without creating the file

**Comment by tiborsimko** [comment]
:+1: for including `check-manifest` in the CI recipe.

@crepererum Can you please also add `include .travis.yml` to the standard `MANIFEST.in` template, otherwise people will have to do it themselves.  For a recent example of a complete `MANIFEST.in`, you can have a look at say https://github.com/inveniosoftware/invenio-collections/blob/master/MANIFEST.in - e.g. there is also `babel.ini` and friends to add.  It would be nice if by default the template comes as complete as possible.

**Comment by crepererum** [comment]
The manifest controls the package installation, right? So why do we install Travis files on the system? They are not required by the user.

**Comment by tiborsimko** [comment]
Indeed, this is not required for the user, but the same can be said about more non-production files say related to testing and coverage that only developer needs (and gets via git).  By including everything in the package, as suggested by `check-manifest`, we are kind of "over-including" stuff; but this does not harm anything that much.  That kind of thinking.

My point was that with the current set of templates, the first Travis run would fail on a newly cut module, because `check-manifest` would lead to an error.  The solution is to either (1) include every file that the cookiecutter generates in its default templates, or else (2) modify MANIFEST.in to exclude those unneeded files so that `check-manifest` wouldn't complain and would pass on its first run on an otherwise unmodified module generated by the cookiecutter.   We usually use the first approach, including everything.

**Comment by crepererum** [comment]
rebased.

**Comment by jirikuncar** [comment]
@crepererum can you check tests?

**Comment by crepererum** [comment]
@jirikuncar done.

**Comment by lnielsen** [comment]
Superseded by #25

**Pull request closed without merging**